docs/anomaly-detection.md

```diff
@@ -1 +1,285 @@
+
+As we can see, there is a curve with a really sharp bend. However, as we said before, for an optimal `eps` range we need to bound the plot.
+
+Here we have a bounded curve with a `range_const` roughly equaling 2.5% of the maximum value of the distances. Setting the `range_const` to this value often provides a good starting point. In our case, it is the plot we are looking for:
+
+![Bounded plot](distance_between_samples_bounded.jpg)
+
+If we set the `range_const` even smaller, to roughly 1% of the maximum, we get a parabola like curve, not the sharp elbow we are looking for.
+
+![Plot with bad range_const](small_range_constant.jpg)
+
+Now all we have to do is to create a relatively small range of `eps` values to test:
+
+```python
+range_const = int(floor(np.amax(column_y)*0.025))
+range_eps = range(range_const, (range_const*5)+1, range_const)
+```
+
+Altough you are free to change `stop` and `step` of the `range` function, if we take a look at the bounded plot with the good range constant, it is visible that a higher `stop` won't be necessary. For the `step`, it is really up to your compute power, for our example, going up with a `step` size of `range_const` is OK.
+
+Now that we have a range of `eps` values to test, it is time to see how they fare against each other. We set up a function called `find_ideal_eps` to do the job:
+
+```python
+def find_ideal_eps(X: np.array, min_samples: int, window_size: int, range_eps: list):
+    # Again, we apply sliding windows to our column of values, take note of the number of windows, and
+    # initialize an array that is esentiallt same as min_sample_scores, but for epsilon values.
+    X_windowed = np.lib.stride_tricks.sliding_window_view(x=X, window_shape=window_size, axis=0)
+    (size_0, _, _) = X_windowed.shape
+
+    eps_scores = np.zeros(shape=(1, len(range_eps)))
+
+    # Here, we compute silhouette scores of each eps value and get a sum for all the windows. For the
+    # clustering we use DBSCAN, which means that we will be using our recently found ideal min_samples value,
+    # as the two hyperparameters affect each other.
+    for window in range(size_0):
+        for i in range(len(range_eps)):
+            clustering = DBSCAN(eps=range_eps[i], min_samples=min_samples).fit(X_windowed[window][0][:].reshape(-1,1))
+            labels = clustering.labels_
+            if np.unique(labels).size > 1:
+                silhouette_avg = silhouette_score(X_windowed[window][0][:].reshape(-1,1), labels)
+                eps_scores[0][i] = eps_scores[0][i]+silhouette_avg
+    
+    # We calculate the average and return the ideal eps value.
+    eps_scores = eps_scores / size_0
+
+    ideal_eps = range_eps[np.where(eps_scores.max)[0][0]]
+
+    return ideal_eps
+```
+
+This is what is needed to find anomalies on a given model. Now we need to set up two more functions to get a Slack bot to message us some information and a plot of the anomalies in our model.
```

Comment:
This section has a lot of information. So we need to make a summary of steps that were taken.